### PR TITLE
Add host and port to timed out error properties

### PR DESCRIPTION
**Pull request by ruimarinho**
Allows for easier debugging when a timeout occurs due to a timed out error. Based on https://github.com/request/request/pull/1747.

### CODE REVIEW

File: request.js [hunk 1]
```diff
@@ -778,6 +778,8 @@ Request.prototype.start = function () {
           var e = new Error('ESOCKETTIMEDOUT')
           e.code = 'ESOCKETTIMEDOUT'
           e.connect = false
+          e.host = self.host
+          e.port = self.port
```
Comment (ruimarinho) [inline request.js]: @FredKSchott any idea why `self.port` is kept as string after url parse? I had to cast the test to string.

Comment (FredKSchott) [inline request.js]: That is bizarre, it looks like it's a number everywhere else. See https://github.com/seegno-forks/request/blob/88f9a04572c2fec21bf46ab2628950c3fae8bf20/request.js#L303-L314

Comment (FredKSchott) [inline request.js]: Ah no, url.parse returns port as a string. But we're setting it manually as a number in other places . 

We should change to always be a number or string in a future PR.

File: tests/test-timeout.js [hunk 2]
```diff
@@ -65,6 +65,20 @@ tape('should set connect to false', function(t) {
...
+tape('should set host and port', function(t) {
+  var shouldTimeout = {
+    url: s.url + '/timeout',
+    timeout: 100
+  }
+
+  request(shouldTimeout, function(err, res, body) {
+    checkErrCode(t, err)
+    t.ok(err.host === 'localhost', 'Read Timeout Error should set \'host\' property')
+    t.ok(err.port === String(s.port), 'Read Timeout Error should set \'port\' property')
```
Comment (FredKSchott) [inline tests/test-timeout.js]: use the actual value of s.port here (`'61160'`), similar to how you did on L76. That will make the test more accurate and solve your casting problem.

File: request.js (hunk 1 above)
Comment (ruimarinho) [inline request.js]: Yep, I haven't spent much time reading the code but it may have introduced a bug already.

File: tests/test-timeout.js (hunk 2 above)
Comment (ruimarinho) [inline tests/test-timeout.js]: The port is dynamic so that's not possible :/

**Comment by franciscocardoso** [comment]
Hi @FredKSchott,

Is there any plans to merge this PR in the near future?

**Comment by FredKSchott** [comment]
@franciscocardoso just rebased. If travis passes, I'm good to merge this.

**Comment by FredKSchott** [comment]
Huh, looks like tests aren't passing anymore. @ruimarinho if you can get tests passing again I'm 👍to merge

**Comment by ruimarinho** [comment]
This is sitting at an unsupported fork right now. Will open a new PR.

**Pull request closed without merging**